docs/docs/examples/vault.md

```diff
@@ -0,0 +1,264 @@
+The primary and security-relevant encryption mechanism remains Vault’s own sealing process, which provides cryptographic protection of
+secrets—even in the event that the underlying storage is compromised.
+
+Because the `workload-secret-seed` is derived from the associated `workloadSecretID`,
+any change to the `workloadSecretID` after the block device has been initialized will result in deriving an invalid encryption key,
+making the mounted block device undecryptable.
+
+:::note[Inter-deployment communication]
+
+The Contrast Coordinator issues mesh certificates after successfully validating workloads.
+These certificates can be used for secure inter-deployment communication.
+The Initializer sends an attestation report to the Coordinator,
+retrieves a service mesh certificate bound to it's provided public key, containing the certificate chain, as well as the current mesh CA cert.
+The Initializer then writes them to a `volumeMount`, allowing to build up the secure mTLS connections based on the service mesh.
+The received service mesh certificate also holds the certificate extension of the `workloadSecretID`,
+which is used to allow the authorization to a certain encryption key on the transit engine API.
+
+:::
+
+
+The Vault's TCP listener is configured to accept connections only from mesh certificates issued under the current state of the service mesh CA,
+effectively restricting communication to attested Contrast deployments.
+The Coordinator’s transit secrets engine API authorizes requests based on the `workloadSecretID`,
+which is embedded in a certificate extension and must match the target endpoint.
+As previously noted, updating the `workloadSecretID` after initializing the LUKS device will make it inaccessible,
+due to a mismatch in the derived encryption key.
+Therefore, it's critical to ensure that the `workloadSecretID` is correctly aligned with the intended endpoint
+specified in Vault’s sealing configuration before the first `contrast set` is executed.
+
+
+### Connecting to the application
+
+Other confidential containers can securely connect to the Vault server via the
+[Service Mesh](../components/service-mesh.md).
+As previously noted, access to the Vault endpoint is restricted to peers that present a service mesh certificate valid
+under the currently set manifest. While such a certificate enables mTLS-based communication
+with the Vault server, it doesn't, on its own, grant authorization to perform Vault-related operations.
+Permissions for accessing secrets within Vault must be explicitly configured using the root token obtained during Vault initialization
+The configured `openbao-client` deployment is responsible for executing Vault-related operations,
+including initialization, secret creation, and sealing instructions.
+
+For more information on the Vault management and administration, please follow the official [OpenBao documentation](https://openbao.org/docs/).
+
+
+## Updating the deployment
```

Comment:
I believe this section could be replaced with a demo snippet that restarts Vault and observes the auto-unsealing (e.g. `kubectl rollout restart` followed by `kubectl logs`).